features/onboarding/passkey/useCreatePasskey.ts

```
@@ -0,0 +1,69 @@
+import { useCallback, useEffect } from "react";
+import { usePasskeyAuthStoreContext } from "./passkeyAuthStore";
+import { usePrivy } from "@privy-io/expo";
+import { useSignupWithPasskey } from "@privy-io/expo/passkey";
+import { usePrivyAuthStoreContext } from "../Privy/privyAuthStore";
+import { captureErrorWithToast } from "@/utils/capture-error";
+import { RELYING_PARTY } from "./passkey.constants";
+
+/**
+ * This hook is used to create a new Passkey account and set the account id in the Passkey Auth Store
+ * It is expected to be used with usePrivySmartWalletConnection, but does not include that hook as this will likely be used with login with passkey
+ * And we would run into issues with effects with how Privy is setup
+ * @returns {
+ *  createPasskey: () => Promise<void>
+ * }
+ */
+export const useCreatePasskey = () => {
+  // Privy Hooks
+  const { logout, user: privyUser } = usePrivy();
+
+  const { signupWithPasskey, state: signupState } = useSignupWithPasskey();
+
+  // Passkey Store Hooks
+  const { setLoading, setStatusString, setError } = usePasskeyAuthStoreContext(
+    (state) => state
+  );
+
+  // Privy Auth Store Hooks
+  const setPrivyAccountId = usePrivyAuthStoreContext(
+    (state) => state.setPrivyAccountId
+  );
+
+  useEffect(() => {
+    setStatusString(signupState.status);
+  }, [setStatusString, signupState.status]);
+
+  const handleCreateAccountWithPasskey = useCallback(async () => {
+    try {
+      setLoading(true);
+      if (privyUser) {
+        // We support multiple accounts, so we need to logout of the current before creating a new one
+        await logout();
```

I still think we should throw here instead of implicitly logging someone out as this just feels confusing and like a hidden side effect

I would throw an error and require that the collar ensures that there's no logged in PRIVY user at the time this is called

---

I don't think we should do this
There's times where we wouldn't want to actually logout the user until they click to create the passkey
For example:

1. Log in on account A
2. Click to add a new account, don't create a new account.
3. User would expect to still be logged in to account B

---

after more context - mainly `privy` assumes one "user" per install which does not meet our usecase so we'll need to manage switching user to our intended user's `privy` account.